server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -494,6 +494,10 @@ public boolean isSystem() {
         return indexSettings.getIndexMetadata().isSystem();
     }
 
+    public CodecService codecService() {
```

Comment:
Can this become `getCodecName()` since it looks like the only thing this is used for is to get the name?